db/upgrade.php

```diff
@@ -341,5 +341,35 @@ function xmldb_cms_upgrade($oldversion) {
         upgrade_mod_savepoint(true, 2023112100, 'cms');
     }
 
+    if ($oldversion < 2023120101) {
+        // Update query for setting.
+        $sql = "UPDATE {customfield_data} mcd1
```

Comment:
Can you please describe what are we doing in this SQL? I'm trying to understand why is it so complicated.

How can we be sure it works as intended? 

Would it work for all DB types?

Reply:
I tested and it's working only Postgres.

I updated the codes to use simple select SQL and update by moodle function.
This should work any DB types.